src/main/java/teammates/logic/core/DataBundleLogic.java

```diff
@@ -109,18 +110,226 @@ public void persistDataBundle(DataBundle dataBundle) throws InvalidParametersExc
         processResponsesAndPopulateMap(responses, sessionResponsesMap);
         processSessionsAndUpdateRespondents(sessions, courseInstructorsMap, sessionQuestionsMap, sessionResponsesMap);
 
-        accountsDb.putEntities(googleIdAccountMap.values());
-        profilesDb.putEntities(profiles);
-        coursesDb.putEntities(courses);
-        instructorsDb.putEntities(instructors);
-        studentsDb.putEntities(students);
-        fbDb.putEntities(sessions);
+        List<AccountAttributes> newAccounts = accountsDb.putEntities(googleIdAccountMap.values());
+
+        List<StudentProfileAttributes> newProfiles = profilesDb.putEntities(profiles);
+        List<CourseAttributes> newCourses = coursesDb.putEntities(courses);
+        List<InstructorAttributes> newInstructors = instructorsDb.putEntities(instructors);
+        List<StudentAttributes> newStudents = studentsDb.putEntities(students);
+        List<FeedbackSessionAttributes> newFeedbackSessions = fbDb.putEntities(sessions);
 
         List<FeedbackQuestionAttributes> createdQuestions = fqDb.putEntities(questions);
         injectRealIds(responses, responseComments, createdQuestions);
 
-        frDb.putEntities(responses);
-        fcDb.putEntities(responseComments);
+        List<FeedbackResponseAttributes> newFeedbackResponses = frDb.putEntities(responses);
+        List<FeedbackResponseCommentAttributes> newFeedbackResponseComments = fcDb.putEntities(responseComments);
+
+        updateDataBundleAccounts(dataBundle, newAccounts);
+
+        updateDataBundleProfiles(dataBundle, newProfiles);
+        updateDataBundleCourses(dataBundle, newCourses);
+        updateDataBundleInstructors(dataBundle, newInstructors);
+
+        updateDataBundleStudents(dataBundle, newStudents);
+        updateDataBundleFeedbackSessions(dataBundle, newFeedbackSessions);
+        updateDataBundleFeedbackQuestions(dataBundle, createdQuestions);
+        updateDataBundleFeedbackResponses(dataBundle, newFeedbackResponses);
+        updateDataBundleFeedbackResponseComments(dataBundle, newFeedbackResponseComments);
```

Comment:
All the logic above looks the same? Why not writing a generic function?

Reply:
Updated all updateXXX methods into a generic function